src/ssvc/decision_points/base.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python
+"""
+file: decisionpoints
+author: adh
+created_at: 9/20/23 10:07 AM
+"""
+
+from dataclasses import dataclass
+from typing import Tuple
+
+from dataclasses_json import dataclass_json
+
+from ssvc._mixins import _Base, _Keyed, _Namespaced, _Versioned
+
+
+## notes
+# based on https://gist.githubusercontent.com/sei-vsarvepalli/de2cd7dae33e1e1dc906d50846becb45/raw/2a85d08a4028f6dd3cc162d4ace3509e0458426f/Exploitation.json
+
+
+@dataclass_json
+@dataclass(kw_only=True)
+class SsvcDecisionPointValue(_Base, _Keyed):
+    """
+    Models a single value option for a decision point.
+    """
+
+    pass
+
+
+@dataclass_json
+@dataclass(kw_only=True)
+class SsvcDecisionPoint(_Base, _Keyed, _Versioned, _Namespaced):
```

Comment:
SsvcDecisionPoints now inherit from _Keyed so they have a key.

Reply:
For JSON Schema creation and automation. I am using the extraneous `field` method with a default metadata and an extension `schemaprops` all defined in base.py  as part of the setup. The _mixin.py also has the similar setup to help automate the creation of JSON schema doc

https://github.com/ahouseholder/SSVC/blob/89daecb74a6289c890dfa8799837b3304921f370/src/ssvc/_mixins.py#L19

https://github.com/ahouseholder/SSVC/blob/89daecb74a6289c890dfa8799837b3304921f370/src/ssvc/decision_points/base.py#L43-L54 

As dataclasses_jsonschema is a bit bloated for this application, our own method is created in doc_helpers/tools.py. I cannot see if we are using GitHub actions to create and automate all the files, so for now, I have just hand created then. The workflow can be updated so the docs/reference/schema/ is entirely automated and is not part of any manual creation. As we are unlikely to change this schema too often it is a bit more than we need to do at this time.

https://github.com/ahouseholder/SSVC/blob/89daecb74a6289c890dfa8799837b3304921f370/src/ssvc/doc_helpers/tools.py#L76